Curator setup as a bosh job

cloud-dude opened the pull request.

Commit:
Errand that purge logs with curator

Commit:
Curator job now has the capability to add/remove purge_logs into crontab.

Commit:
clean curator_ctl start and stop scripts

Commit:
Describe curator capabilities in readme

dpb587 commented on templates/logsearch-infrastructure-warden.yml:
```diff
@@ -83,417 +83,417 @@ networks_default:
   subnets:
   - cloud_properties:
       name: random
-    range: 10.244.0.0/30
```
Does this file really need to change for this PR?

cloud-dude replied:
Well we did this because it collided with IPs used by cf-warden. We did all our development in bosh-lite thats why we had to change the templates so that we can have CF, L4CF and Logsearch in the same infrastructure.

dpb587 replied:
Makes sense - hadn't seen a reference to this in the commits and thought it might be unintentionally included.

cloud-dude replied:
You are right on that one, we need to get better at describing our commits for these bigger PRs

dpb587 commented on templates/logsearch-jobs.yml:
```diff
@@ -26,6 +29,8 @@ properties:
     host: (( jobs.queue.networks.default.static_ips.[0] ))
   api:
     elasticsearch_host: (( jobs.api.networks.default.static_ips.[0] ":9200"))
+  purge_logs:
```
Is this purge_logs a rename artifact that can now be removed?

cloud-dude replied:
You are right on this one, we will amend the code.

dpb587 commented:
Please remove the blobs from src/ and squash them out of existence from this PR. I've cloned them locally so I can upload them to the public bucket, but won't merge the PR while they still exist in git history.

dpb587 commented on jobs/curator/templates/bin/curator_ctl:
```diff
@@ -0,0 +1,50 @@
+case $1 in
+  start)
+    if [ $CRON_STATUS -eq 0 ]; then
+      /usr/sbin/cron start
```
I believe cron is already running on all standard stemcells - BOSH pre-configures it for both NTP synchronization and log rotation. I don't think we should be concerning ourselves with starting it, especially if we're not taking responsibility for shutting it down if we did start it.

cloud-dude replied:
While working on bosh-lite we experienced that cron did not start, that is why we added this logic.

dpb587 replied:
Interesting - wasn't aware of that issue. Thanks for explaining. I guess it'll be easiest to leave it for now, then.

dpb587 replied:
Just randomly found [this recent story about enabling cron on bosh-lite](https://www.pivotaltracker.com/story/show/101179464) while working on something else. Guess after that happens we can simplify this control script later.

dpb587 commented on jobs/curator/templates/bin/curator_ctl:
```diff
@@ -0,0 +1,50 @@
+      echo "$CRON_SCHEDULE" >> /tmp/crontab_buffer
+    fi
+
+    crontab < /tmp/crontab_buffer
```
This whole approach is kind of fragile with the listing/grepping/appending on terms. Did you have a look at how the [archiver](https://github.com/logsearch/logsearch-boshrelease/blob/develop/jobs/archiver/templates/bin/logstash.process.erb#L49) installs a cron task - it's much simpler and, as far as I know, hasn't been a problem for anybody. If you did have problems, I'd be interested to hear and fix (and in helping come up with a less fragile approach than this one).

dpb587 commented on jobs/curator/templates/bin/purge_logs:
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```
If this were renamed to just `run`, I think this job could have additional value as being errand-friendly which may work better in some users' BOSH lifecycles.

dpb587 commented:
Thanks for sending this! I'm excited to merge and be able to use this in our own clusters and deprecate our Jenkins job. I've left a few line/regular comments as I reviewed it - let me know if you have any questions!

dpb587 commented:
Looks like you might have already squashed the binaries out. I uploaded the blobs and you or I will want to include the following eventually during merge...

```
diff --git a/config/blobs.yml b/config/blobs.yml
index b3cf3f8..ff79c63 100644
--- a/config/blobs.yml
+++ b/config/blobs.yml
@@ -43,3 +43,23 @@ elasticsearch/elasticsearch-1.7.1.tar.gz:
   object_id: 5ba2c3ed-1730-40fc-9646-aafcab7d9c39
   sha: 0984ae27624e57c12c33d4a559c3ebae25e74508
   size: 28500556
+curator/click-3.3.tar.gz:
+  object_id: 5a17f7ca-b4dc-467a-b5a3-9181dfba17b6
+  sha: d716a932b930d71059e49465b6b42e833808369a
+  size: 259855
+curator/elasticsearch-curator.tar.gz:
+  object_id: dbacd365-5287-439b-8d48-006be450ee5b
+  sha: 28fd2f073b92b33e01db0df08adfc16699969e81
+  size: 82745
+curator/elasticsearch-py.tar.gz:
+  object_id: 44bd1583-2ea7-463d-97f8-a0860da085eb
+  sha: d6d61ddb5c20b0f806978174f75754bce1c53e41
+  size: 77205
+curator/urllib3-1.8.3.tar.gz:
+  object_id: f5601b3b-e555-4ca0-b955-c01e4081d0ee
+  sha: 00c92b0c3b0756ae3c4e6262d5bc80023dfe9ca3
+  size: 85466
+python/Python-3.4.3.tgz:
+  object_id: 34ff2b80-c2ca-4075-81a7-164805059d89
+  sha: 94cae02a334c72ea6ba8c056bb2864e2838878d3
+  size: 19554643
```

Commit:
Refactor curator job

- Start and stop script in curator_ctl now do not relay on temp files.
- Created cron config for purge logs within the config folder.
- Rename monit job for curator (curator_ctl -> curator)

cloud-dude commented:
@dpb587 Take a look at the last refactor, if you need any additional changes please let us know. Would you mind doing the merge of the `config/blobs.yml`?

@bonzofenix  && @cloud-dude

dpb587 commented:
Sure thing; will take a look soon and @you if anything else comes up.

dpb587 commented:
Merged - thanks!
